zerver/actions/message_send.py

```diff
@@ -1122,6 +1124,19 @@ class UserData(TypedDict):
             if send_request.stream.first_message_id is None:
                 send_request.stream.first_message_id = send_request.message.id
                 send_request.stream.save(update_fields=["first_message_id"])
+
+            if user_access_restricted_in_realm(
+                send_request.message.sender
+            ) and not subscribed_to_stream(send_request.message.sender, send_request.stream.id):
```

Comment:
We only allow sending a single at a time message using API and we only send multiple messages in development environment and probably for some notifications sent by bots, so this should not really do query in a loop. For bots the `user_access_restricted_in_realm` will anyways be `False` without doing any query.

Reply:
OK -- let's add a comment though.